x/wasm/internal/keeper/ioutil.go

```diff
@@ -28,6 +30,24 @@ func uncompress(src []byte) ([]byte, error) {
 		return nil, err
 	}
 	zr.Multistream(false)
+	defer zr.Close()
+	return ioutil.ReadAll(LimitReader(zr, int64(limit)))
```

Comment:
This means if the total size >= limit we will get the types.ErrLimit error. However, we want total_size > limit.
I would use `int64(limit + 1)` here, so if there is one byte more than the requested limit, it returns an error